src/Microsoft.AspNet.Security.OpenIDConnect/OpenIdConnectAuthenticationOptions.cs

```
@@ -0,0 +1,337 @@
+        public OpenIdConnectProtocolValidator ProtocolValidator
+        {
+            get
+            {
+                return _protocolValidator;
+            }
+            set
+            {
+                if (value == null)
+                {
+                    throw new ArgumentNullException("value");
+                }
+
+                _protocolValidator = value;
+            }
+        }
+
+        /// <summary>
+        /// Gets or sets the 'post_logout_redirect_uri'
+        /// </summary>
+        /// <remarks>This is sent to the OP as the redirect for the user-agent.</remarks>
+        [SuppressMessage("Microsoft.Design", "CA1056:UriPropertiesShouldNotBeStrings", Justification = "By design")]
+        [SuppressMessage("Microsoft.Naming", "CA1726:UsePreferredTerms", MessageId = "Logout", Justification = "This is the term used in the spec.")]
+        public string PostLogoutRedirectUri { get; set; }
+
+        /// <summary>
+        /// Gets or sets the 'redirect_uri'.
+        /// </summary>
+        [SuppressMessage("Microsoft.Design", "CA1056:UriPropertiesShouldNotBeStrings", Justification = "By Design")]
+        public string RedirectUri { get; set; }
+
+        /// <summary>
+        /// Gets or sets the 'resource'.
+        /// </summary>
+        public string Resource { get; set; }
+
+        /// <summary>
+        /// Gets or sets the 'response_type'.
+        /// </summary>
+        public string ResponseType { get; set; }
+
+        /// <summary>
+        /// Gets or sets the 'scope'.
+        /// </summary>
+        public string Scope { get; set; }
+
+        /// <summary>
+        /// Gets or sets the AuthenticationType used when creating the <see cref="System.Security.Claims.ClaimsIdentity"/>.
+        /// </summary>
+        public string SignInAsAuthenticationType
+        {
+            get { return TokenValidationParameters.AuthenticationType; }
```

This kind of voodoo magic proved to offer a really bad user experience in the past: if you set it prior to setting `TokenValidationParameters`, the value set to SignInAsAuthenticationType will be silently ignored.